src/libasr/pass/array_op.cpp

```diff
@@ -978,6 +978,8 @@ void pass_replace_array_op(Allocator &al, ASR::TranslationUnit_t &unit,
     std::string rl_path = pass_options.runtime_library_dir;
     ArrayOpVisitor v(al, rl_path);
     v.visit_TranslationUnit(unit);
+    PassUtils::UpdateDependenciesVisitor u(al);
+    u.visit_TranslationUnit(unit);
```

Comment:
Looks like we now have these extra passes that need to be run.

Reply:
There are two choices with us,

1. Either track dependencies while writing a pass. So a pass involves first doing the job of writing the visitor like in the above case `ArrayOpVisitor` and then track dependencies in the pass itself i.e., inside `ArrayOpVisitor`.
2. Write the pass visitor `ArrayOpVisitor` and then call `PassUtils::UpdateDependenciesVisitor`.

The second approach decouples writing a pass and tracking dependencies. Calling `PassUtils::UpdateDependenciesVisitor` is not necessary after every calling every pass visitor, just only those which add functions, structures or modify variables. Passes of such kind are very few in number AFAIK.

Reply:
Ok!